dartagnan/src/main/java/com/dat3m/dartagnan/verification/solving/RefinementSolver.java

```diff
@@ -205,6 +208,12 @@ private void runInternal(SolverContext ctx, ProverWithTracker prover, Verificati
         Context baselineContext = Context.createCopyFrom(analysisContext);
         performStaticWmmAnalyses(task, analysisContext, config);
 
+        // Encoding context with the original Wmm for relation extraction.
+        EncodingContext contextWithFullWmm = EncodingContext.of(
+            task, analysisContext, ctx.getFormulaManager()
+        );
+        setContextWithFullWmm(contextWithFullWmm);
+
```

Comment:
Why is this needed?

Reply:
Ok, I should be more clear. (1) Why is this used in the `WMMSolver` and (2) why do you need a whole encoding context, if all you want is the full `Wmm`? You could, for example, store the original `VerificationTask` instead.

Reply:
(1)None of the solving methods in `WMMSolver` uses the new model. It just hold two models and give the right one to `RefinementSolver`. For example give the old one to the analysis and solving methods and give the new one to `generateGraphvivFiles` inside `RefinementSolver`. (2) a full `Wmm` is not enough because sometimes the `EncodingContext.EdgeEncoder` is needed to make the relation extraction more accurate.

Reply:
So you say that `WmmSolver` does not even need the new model because it doesn't use it. Only the `RefinementSolver` needs it for witness generation. Then you can avoid the detour and just let the `RefinementSolver` directly generate a separate `ExecutionModelNext` (this is what I was referring to in the second comment about the changes to the old `ExecutionModel`)